src/Plugin.php

```diff
@@ -96,6 +108,8 @@ function (Result $result) use (&$totals): void {
 
                 $color = $uncoveredLines === [] ? 'green' : 'yellow';
 
+                $this->coverageLogger?->append( $path, $uncoveredLines, $uncoveredLinesIgnored, $result->totalCoverage );
```

Comment:
Can you use a Null Logger implementation, so we don't have to apply conditionals like `?->`?

Reply:
Done, great ideas! it is now instantiated during Plugin construction by default, I am not sure if this could be approached better?

Moved to Contracts namespace as well!